src/openapi-generator.ts

```diff
@@ -473,10 +483,19 @@ export class OpenAPIGenerator {
       },
     };
 
-    this.pathRefs[path] = {
-      ...this.pathRefs[path],
-      ...routeDoc,
-    };
+    if (type === 'route') {
+      this.pathRefs[path] = {
+        ...this.pathRefs[path],
+        ...routeDoc,
+      };
+    }
+
+    if (type === 'webhook') {
```

Comment:
How do you feel about a switch/case on this without a `default` case so that TS can verify if it's exhaustive or not? Though in this case it won't complain in both cases but still feels better.